apps/blog/_posts/think-before-you-code.md

```diff
@@ -0,0 +1,55 @@
+
+## Porque é que é importante pensar antes de escrever código?
+
+Existem várias razões. Em primeiro lugar, programar pode ser um processo complexo e demorado. Se começares a escrever código sem um plano claro em mente, provavelmente perderás muito tempo com falsas partidas, becos sem saída e _debugging_. Ao reservar um tempo para pensar nas coisas de antemão, **economizas muito tempo** e **poupas frustração** a longo prazo.
+
+Outro aspeto importante de pensar antes de programar é entender a matemática por trás da programação. Embora linguagens de programação como **_Python_, _Java_, _C_**… nos permitam criar programas poderosos, elas são construídas com base em conceitos matemáticos. Quando tens um plano claro em mente, é menos provável que cometas erros ou ignores detalhes importantes. Por exemplo, os algoritmos, que são uma parte fundamental do estudo desta área, são baseados em princípios matemáticos como **eficiência**, **correção** e **escalabilidade**.
+
+Por fim, pensar antes de bater no teclado ajuda-te a ser um programador melhor, em geral. Quando reservas algum tempo para planear o teu código, és forçado a pensar mais profundamente sobre o problema que resolves. Isso pode levar a soluções mais elegantes, a uma melhor organização de código e algoritmos mais eficientes. Pensar num problema, é também considerar casos extremos e possíveis erros. Considerar esses cenários antes de escrever o código, ajuda-te a ter uma solução mais robusta e menos propensa a erros. Na programação, é essencial **considerar todos os resultados** possíveis e garantir que o teu código esteja preparado para lidar com eles. Desenvolver bons hábitos de planeamento desde o início, preparara-te para o **sucesso** enquanto continuas a aprender e crescer como programador.
+
+## Resumo
+
+Então, fazendo as contas finais, as **principais ideias** são:
+
+1. **Define o problema:** antes de começar a programar, certifica-te de entender exatamente qual problema tentas resolver. Faz uma **descrição clara do problema**, com quaisquer restrições ou requisitos que precisas considerar.
+
+2. **Divide:** após entender o problema, divide o problema em **partes mais pequenas e mais simples**. Pensa nas etapas que precisas seguir para resolver cada peça e como elas se encaixam para formar a solução maior.
+
+3. **Planeia os teus algoritmos**: Se o problema envolve cálculos matemáticos ou algoritmos, reserva um tempo para planear. Escreve as etapas que precisas seguir e certifica-te de entender a lógica por trás de cada etapa, como um **verdadeiro matemático**.
+
+4. **Fazer desenhos:** se ajudar, desenha diagramas ou fluxogramas para **ajudar a visualizar** a solução. Isso pode ser especialmente útil para problemas ou algoritmos complexos.
+
+5. **Testar**: Ainda antes de passares para o computador **confirma** que tens a solução correta! Usa a tua cabeça ou até mesmo escreve um _debugger_ por exemplo um pseudocódigo para garantir que esteja tudo a funcionar conforme o esperado.
```

Comment:
I think it would be nice to mention [TDD](https://pt.wikipedia.org/wiki/Test-driven_development) as a fun fact here 👀